std/conv.d

```diff
@@ -2032,6 +2034,49 @@ if (is(T == enum) && !is(S == enum)
     assert(m1 == [[En8143.A, En8143.C], [En8143.C, En8143.A]]);
 }
 
+// issue 20539
+@safe pure unittest
+{
+    // needs to be static, because else a frame pointer issue makes
+    // it work unintentionally; see next unittest for the same test
+    // with the struct being not static
+    static struct A
+    {
+        auto opEquals(U)(U)
+        {
+            return true;
+        }
+    }
+
+    enum Color
+    {
+        red = A()
+    }
+
+    import std.exception : assertNotThrown;
+    assertNotThrown("xxx".to!Color);
+}
+
+// issue 20539
+@safe pure unittest
```

Comment:
Is there a good reason to split this into 2 unittest instead of e.g. 2 scopes? That avoids the duplicate import and unittest declaration

Reply:
Don't know. It feels a little bit like two different things. The first one guards against the bug, while the second one guards against not introducing a new bug with the bug fix. On the other hand, they are quite similar. Do you mind me leaving it, like it is?

Reply:
No, it's mostly a matter of preference (except for a possible but negligible difference in compile/run time)